index.js

```diff
@@ -532,9 +558,17 @@
 
   //  Number$prototype$equals :: Number ~> Number -> Boolean
   function Number$prototype$equals(other) {
-    return typeof this === 'object' ?
-      equals(this.valueOf(), other.valueOf()) :
-      isNaN(this) && isNaN(other) || this === other && 1 / this === 1 / other;
+    return typeof other === 'object' ?
+      typeof this === 'object' &&
+        equals(this.valueOf(), other.valueOf()) :
+      isNaN(other) && isNaN(this) || this === other;
+  }
+
+  //  Number$prototype$lte :: Number ~> Number -> Boolean
+  function Number$prototype$lte(other) {
+    return isNaN(this) ? isNaN(other) :
+      (typeof this === 'object' ? this.valueOf() : this)
+        <= (typeof other === 'object' ? other.valueOf() : other);
   }
```

Comment:
Among others, this now breaks both these tests:

```
eq(Z.lte(42, new Number(42)), true);
eq(Z.lte(new Number(42), 42), true);
```

Are we _sure_ it's worthwhile to differentiate here? It does go back to the point that we're making a boolean type with 4 different constructors, and now 2 * ∞(ish) numbers. If the above tests _break_, and this distinction is maintained, totality is broken.

Alternatively, we then have to pick whether `x <= new Number(x)` or `x >= new Number(x)`, and the same for other types such as (but not limited to) `Boolean`. That behaviour would be pretty unintuitive, because I then have a `false` value that may not be `<=` another `false`, depending on where I get those values.

It seems as though we either have to treat both forms of the same value as the same value, or not support constructor-borne primitives. What do you think?